u2f-ref-code/java/src/com/google/u2f/server/ControlFlags.java

```diff
@@ -0,0 +1,87 @@
+package com.google.u2f.server;
+
+/**
+ * If later versions add more fields, this object can store the parsed control byte data. This can
+ * be extended to an arbitrary number of fields, so that many control bytes can be chained together
+ * to allow the server to process many different types of messages.
+ */
+public class ControlFlags {
+  private final boolean userPresence;
+  private final boolean isTransferAccessMessage;
+
+  /**
+   * Hidden constructor, use builder instead
+   *
```

Comment:
comments don't line up with implementation.  (you mention 2 params, but there is only one).

I would go with the model that you have in the javadoc rather than passing a whole builder in.